### PR TITLE
feature: allow use of short lived creds for local container

### PR DESCRIPTION
**Pull request by wcarpenter1-godaddy**
*Issue #, if available:*
https://github.com/aws/sagemaker-python-sdk/issues/3464

*Description of changes:*
This change allows the user to tell the sagemaker local container to use short-lived credentials in the session.  This is necessary if you would like to use local mode on an ec2 machine and maintain the session's creds for which you have assumed a role, instead of having SM automatically use the ec2 machine's metadata credentials.  

Our use case is that we run Sagemaker [local](https://docs.aws.amazon.com/sagemaker/latest/dg/pipelines-local-mode.html) endpoint deploy during our Github Actions CI/CD to verify that the Sagemaker endpoint is valid before initiating an actual deploy.  We are using a custom container for the endpoint.  The container has code to fetch AWS secrets from the assumed-role in the GHA session.  Since SM currently forces the container to use metadata credentials, we are unable to find the correct secrets, and so our local deploy fails.

By allowing SM to use the assumed-role session's credentials when operating on and EC2, it should allow our local deploy to work as expected.

*Testing done:*
I've installed the forked repo's changes on our project and it fixed the issue outlined above.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by wcarpenter1-godaddy** [comment]
@jmahlik could this code change be a work around for the problem you are seeing here https://github.com/aws/sagemaker-python-sdk/issues/3464?

**Comment by jmahlik** [comment]
> @jmahlik could this code change be a work around for the problem you are seeing here #3464?

Yes, this would fix the issue perfectly :). Seems reasonable to me having it as an env var.

I think there might need to be some more documentation on setting the env var. I don't think users would be able to find this option without going through the source code.

Another thought I had was making this an option to the `LocalSession`'s config. But that would require re-wiring the session and how it passes the config to this function ect so I gave up and just passed the keys via environment variables to the local job.

**Review by navinsoni — COMMENTED** [review]
/bot run all

**Review by navinsoni — COMMENTED** [review]
/bot run all

**Comment by wcarpenter1-godaddy** [comment]
@navinsoni thanks for helping run the checks.  what is the process for getting this PR merged?

**Review by claytonparnell — COMMENTED** [review]
/bot run slow-tests

File: src/sagemaker/local/image.py [hunk 1]
```diff
@@ -1052,6 +1052,10 @@ def _aws_credentials_available_in_metadata_service():
     return not instance_metadata_provider.load() is None
 
 
+def _use_short_lived_credentials():
```
Comment (claytonparnell) [inline src/sagemaker/local/image.py:1055]: Pls add docstring to function

**Comment by claytonparnell** [comment]
Hi @wcarpenter1-godaddy , I apologize Navin went on parental leave and nobody picked this up; I can help get it merged, just need a few things. Could you please:
1. Add docstring to `_use_short_lived_credentials`
2. resolve merge conflict in test_image.py
3. rebase your branch 
And then we should be good to go!

**Comment by wcarpenter1-godaddy** [comment]
Thanks @claytonparnell should be updated

**Review by claytonparnell — COMMENTED** [review]
/bot run pr, slow-tests

**Comment by wcarpenter1-godaddy** [comment]
@claytonparnell good to merge? or do we need to wait until slow-tests/pr builds succeed?  seem like those also fail on other branches

**Review by claytonparnell — COMMENTED** [review]
/bot run slow-tests

**Comment by jmahlik** [comment]
Maybe in a separate PR, would it be possible to document this somewhere.

**Comment by wcarpenter1-godaddy** [comment]
@jmahlik maybe a new bullet point [here](https://github.com/aws/sagemaker-python-sdk/blob/master/doc/overview.rst?plain=1#L1575)?

**Review by claytonparnell — COMMENTED** [review]
/bot run slow-tests

**Review by claytonparnell — APPROVED** [review]

**Comment by wcarpenter1-godaddy** [comment]
😄 😭 😄 

Thanks @claytonparnell !

**Pull request merged**